### ISSUE TITLE
Since 3.1.0 matcher validate_uniqueness_of fails for integer fields

### ISSUE THREAD

**Issue by obfuscoder** [thoughtbot/shoulda-matchers#885]
I have the following model:

```ruby
create_table "items", force: :cascade do |t|
  ...
  t.integer  "reservation_id"
  t.integer  "number"
  ....

class Item < ActiveRecord::Base
  validates :number,
    numericality: { greater_than: 0, only_integer: true },
    uniqueness: { scope: :reservation_id },
    allow_nil: true
```

After I updated from 3.0.1 to 3.1.0 The following expectation started to fail:

```ruby
RSpec.describe Item do
  subject { build(:item) }
  it { is_expected.to validate_uniqueness_of(:number).scoped_to(:reservation_id) }
```

With the following error:

```
ActiveRecord::RecordInvalid: Validation failed: translation missing: en.activerecord.attributes.item.number is not a number
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/validations.rb:79:in `raise_record_invalid'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/validations.rb:43:in `save!'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/attribute_methods/dirty.rb:29:in `save!'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/transactions.rb:291:in `block in save!'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/transactions.rb:351:in `block in with_transaction_returning_status'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/connection_adapters/abstract/transaction.rb:184:in `within_new_transaction'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/transactions.rb:220:in `transaction'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/transactions.rb:348:in `with_transaction_returning_status'
/home/asi/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.5/lib/active_record/transactions.rb:291:in `save!'
/home/asi/.rvm/gems/ruby-2.2.2/gems/shoulda-matchers-3.1.0/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:494:in `update_existing_record!'
/home/asi/.rvm/gems/ruby-2.2.2/gems/shoulda-matchers-3.1.0/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:583:in `validate_two_records_with_same_non_blank_value_cannot_coexist?'
/home/asi/.rvm/gems/ruby-2.2.2/gems/shoulda-matchers-3.1.0/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:320:in `matches?'
./spec/models/item_spec.rb:12:in `block (2 levels) in <top (required)>'
```

Adding option **ignoring_case_sensitivity** does not help.

**Comment by mcmire** [comment]
This is related to #880 (although not quite the same) and I'll have a fix for this soon.

**Comment by mcmire** [comment]
This should be fixed in 3.1.1.
